macros/bcmacrotest.ejs

```diff
@@ -0,0 +1,275 @@
+      }
+      // Add alternative name
+      if (entries[entry].alternative_name) {
+        entryResult += '(as ' + entries[entry].alternative_name + ')';
+      }
+
+      // Generate notes, if any
+      if (entries[entry].notes != undefined) {
+        if (!Array.isArray(entries[entry].notes)) {
+          entryResult += addNote(noteCollection, entries[entry].notes);
+        }
+        else {
+          for (note in entries[entry].notes) {
+            if (!entries[entry].notes.hasOwnProperty(note)) continue;
+            entryResult += addNote(noteCollection, entries[entry].notes[note]);
+          }
+        }
+      }
+
+      // Add synthetic notes
+      for (var note in syntheticNotes) {
+        if (!syntheticNotes.hasOwnProperty(note)) continue;
+        noteCollection.push(syntheticNotes[note]);
+        entryResult += addNote(noteCollection, syntheticNotes[note]);
+
+      }
+      syntheticNotes = new Array; // empty notes array
+    }
+  }
+  return entryResult;
+}
+
+// Create the top of the top of the table
+var top =  '<div class="htab">\n';
+top += '  <a name="AutoCompatibilityTable" id="AutoCompatibilityTable"></a>\n';
+top += '  <ul>\n';
+top += '    <li class="selected"><a>Desktop</a></li>\n';
+top += '    <li><a>Mobile</a></li>\n';
+top += '  </ul>\n';
+top += '</div>\n';
+
+// Create the top of the desktop div
+var desktop =  '<div id="compat-desktop">\n';
+desktop += '  <table class="compat-table">\n';
+desktop += '    <tbody>\n';
+desktop += '      <tr>\n';
+desktop += '        <th>Feature</th>\n';
+for (var browser in desktopBrowsers) {
+  desktop += '        <th>' + desktopNames[browser] + '</th>\n';
+}
+desktop += '      </tr>\n';
+
+// Create the top of the mobile div
+var mobile =  '<div id="compat-mobile">\n';
+mobile += '  <table class="compat-table">\n';
+mobile += '    <tbody>\n';
+mobile += '      <tr>\n';
+mobile += '        <th>Feature</th>\n';
+for (var browser in mobileBrowsers) {
+  mobile += '        <th>' + mobileNames[browser] + '</th>\n';
+}
+mobile += '      </tr>\n';
+
+// For each sub-feature, add a line in both the desktop and mobile divs
+for (var subfeat in featureBC.__compat ) {
+   // skip loop if the property is from prototype
+   if (!featureBC.__compat.hasOwnProperty(subfeat)) continue;
+
+   // Add subfeature compat info for desktop
+   desktop += '     <tr>\n';
+   desktop += '       <td>' + featureBC.__compat[subfeat].desc + '</td>\n';
+   for (var browser in desktopBrowsers) {
+     desktop += '       <td>' + displayCompatInfo(featureBC.__compat[subfeat].support[desktopBrowsers[browser]], desktopBrowsers[browser]) + '</td>\n';
+   }
+   desktop += '     </tr>\n';
+
+   // Add subfeature compat info for mobile
+   mobile += '   <tr>\n';
+   mobile += '     <td>' + featureBC.__compat[subfeat].desc + '</td>\n';
```

Comment:
The description is optional. You need to fall back to the identifier if it's not present.